### PR TITLE
Add Safari 17 data from the BCD collector

### PR DESCRIPTION
**Pull request by jdatapple**
#### Summary

Adds Safari 17 data from the MDN BCD collector tool.

#### Test results and supporting details

- [Safari 17 Beta Release Notes](https://developer.apple.com/documentation/safari-release-notes/safari-17-release-notes)

### CODE REVIEW

**Comment by Elchi3** [comment]
@jdatapple there is https://github.com/mdn/browser-compat-data/pull/20064 already

**Comment by jdatapple** [comment]
I noted that after I had submitted. This is the same data, however, it also includes the necessary `experimental` flag changed in this version.

File: api/CSSCounterStyleRule.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
           "opera": "mirror",
           "opera_android": "mirror",
           "safari": {
-            "version_added": false,
+            "version_added": "17",
             "notes": "See <a href='https://webkit.org/b/167645'>bug 167645</a>."
```
Comment (Elchi3) [inline api/CSSCounterStyleRule.json:25]:
```suggestion
```

File: api/CanvasRenderingContext2D.json [hunk 2]
```diff
@@ -815,7 +815,7 @@
               "opera": "mirror",
               "opera_android": "mirror",
               "safari": {
-                "version_added": false,
+                "version_added": "17",
                 "notes": "See <a href='https://webkit.org/b/226964'>bug 226964</a>."
```
Comment (Elchi3) [inline api/CanvasRenderingContext2D.json:819]:
```suggestion
```

**Review by Elchi3 — COMMENTED** [review]
Thanks! Copying over my other review comment here as well. The notes with bug references aren't needed anymore

**Pull request closed without merging**